### PR TITLE
[Enhancement] constcolumn ctor check datacolumn size

### LINKED ISSUE #5186: Ensure that the data column size of the const column does not exceed 1 to avoid additional computational overhead

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [x] enhancement

## Which issues of this PR fixes ：
Fixes #5186 

## Problem Summary(Required) ：
For ConstColumn's datacolumn only the first element is useful, but sometime the datacolumn's size is not only one(for example, PercentileFunctions::percentile_hash, VectorizedInConstPredicate::eval_on_chunk_both_column_and_set_not_has_null, HyperloglogFunction::hll_hash, DecimalBinaryFunction::evaluate, ColumnHelper::create_column and ColumnBuilder::build), and the datacolumn;s size will be  used in class UnpackConstColumnUnaryFunction( UnaryFunction, StringUnaryFunction and so on),so check and resize the datacolumn at constcolumn::ctor

### CODE REVIEW

File: be/src/column/const_column.cpp [hunk 1]
```diff
@@ -12,10 +13,17 @@ namespace starrocks::vectorized {
 
 ConstColumn::ConstColumn(ColumnPtr data) : _data(std::move(data)), _size(0) {
     DCHECK(!_data->is_constant());
+    // _data->resize(1);
+    // DCHECK(_data->size() == 1);
 }
 
 ConstColumn::ConstColumn(ColumnPtr data, size_t size) : _data(std::move(data)), _size(size) {
     DCHECK(!_data->is_constant());
+    std::cout << _data->size() << std::endl;
```
Comment (trueeyu) [inline be/src/column/const_column.cpp:22]: remove the log

File: be/src/column/const_column.cpp [hunk 2]
```diff
@@ -12,10 +13,17 @@ namespace starrocks::vectorized {
...
+    if (_data->size() > 1) {
+        _data->resize(1);
+        DCHECK(_data->size() == 1);
```
Comment (trueeyu) [inline be/src/column/const_column.cpp:25]: the DCHECK is useless, because the code above already resize it to 1

Comment (zombee0) [reply be/src/column/const_column.cpp:25]: sorry, i submit a error version without some modify, i'll push a new version after unittest,  wait a moment

**Review by imay — DISMISSED** [review]

**Review by silverbullet233 — DISMISSED** [review]

File: be/src/column/const_column.cpp [hunk 3]
```diff
@@ -12,10 +12,16 @@ namespace starrocks::vectorized {
 
 ConstColumn::ConstColumn(ColumnPtr data) : _data(std::move(data)), _size(0) {
```
Comment (stdpain) [inline be/src/column/const_column.cpp:13]: prefer
```suggestion
ConstColumn::ConstColumn(ColumnPtr data) : ConstColumn(data, 0) {}

```

Comment (zombee0) [reply be/src/column/const_column.cpp]: thank you, it's better

**Review by silverbullet233 — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**